text/0000-PXE-Booting-UEFI.md

```diff
@@ -0,0 +1,228 @@
+consistency, a change in Foreman core will be needed but no migration or manual
+change after upgrade is needed - existing hosts in boot mode will continue
+booting from the old dir, new ones will use the new path. TFTP file deletion
+does not trigger error, thus the only drawback is there might be configuration
+files left after some time. We will put this into upgrade instructions.  Edit
+
+## Implementation
+
+Both Foreman Core and Smart Proxy changes are required in order to support all
+of the above.
+
+### [Foreman proxy #12633](http://projects.theforeman.org/issues/12633) - Add Pxegrub2 variant and allow multiple TFTP config files
+
+Foreman Proxy must be enhanced with new PXE variant "Pxegrub2" to allow
+deploying configuration files in "grub2/" folder. In addition to that, Pxegrub
+(version 1) must deploy two configuration files instead of one now to support
+both BIOS and UEFI naming conventions (grub/menu.lst vs grub/efidefault and
+grub/menu.lst.01aabbccddeeff vs grub/01-AA-BB-CC-DD-EE-FF). Also, Pxegrub
+variant will use new base directory "grub/" instead of "boot/grub/" for
+consistency with "grub2" and "pxelinux.cfg/".
+
+### [Foreman core #12634](http://projects.theforeman.org/issues/12634) - New host/hostgroup flags pxe_template and loader
+
+Host and hostgroups must be enhanced with two new flags: PXE template specifies
+PXE template kind and variant pair to use. Until today, this was hardcoded to
+PXELinux or PXEGrub by selected operating system. Optionally, template variant
+flag from OS could be simply dropped because PXE template implies variant to
+use and there are no variants which share the same configuration syntax (this
+is an open item).
+
+Orchestration TFTP code must be modified to use multiple "KIND LOADER default
+local boot" and "KIND LOADER global default" templates. Local and global boot
+templates must be added for Grub2 new kind. Naming convention will be:
+
+* PXELinux BIOS default local boot
+* PXELinux UEFI default local boot
+* PXEGrub BIOS default local boot
+* PXEGrub UEFI default local boot
+* etc. (the same for global default templates)
+
+But only kind/variant is not enough, users need to select BIOS or UEFI booting
+method. This is where Loader flag comes in - selection of three constants:
+BIOS, UEFI and SecureBoot UEFI. The first two are obvious causing selection of
+the correct DHCP filename when deploying TFTP configuration, the third one
+select "shim.efi" chain loading capability for secure booting.
+
+Similarly to root password, a global setting could be provided for one or both
+flags for missing values (open item).
+
+### [Foreman installer #14920](http://projects.theforeman.org/issues/14920) - Change dhcpd.conf to support EFI loaders
+
+Default DHCPD configuration must be changed in order to allow booting unknown
+UEFI hosts as it was shown above (dhcpd.conf snippet). Using shim.efi is
+perhaps better default than loading Grub directly.
+
```

Comment:
i was just referring to: _perhaps better default than loading Grub directly_ - so i was reading it as enable shim by default - if it makes any sense?